pkg/gitreceive/k8s_util.go

```diff
@@ -204,21 +204,19 @@ func waitForPodEnd(c *client.Client, ns, podName string, interval, timeout time.
 		return false, nil
 	}
 
-	return waitForPodCondition(c, ns, podName, condition, interval, timeout)
+	return waitForPodCondition(pw, ns, podName, condition, interval, timeout)
 }
 
 // waitForPodCondition waits for a pod in state defined by a condition (func)
-func waitForPodCondition(c *client.Client, ns, podName string, condition func(pod *api.Pod) (bool, error),
+func waitForPodCondition(pw *k8s.PodWatcher, ns, podName string, condition func(pod *api.Pod) (bool, error),
```

Comment:
can you adapt `NamespaceWatcher` so that `*k8s.PodWatcher` conforms to it, and then make this `pw NamespaceWatcher`?